src/algorithm/center-of-mass.hpp

```diff
@@ -221,6 +221,46 @@ namespace pinocchio
                        DataTpl<Scalar,Options,JointCollectionTpl> & data,
                        const bool computeSubtreeComs = true);
 
+  ///
+  /// \brief Computes the jacobian of the center of mass of the given subtree according to a particular joint configuration.
+  ///        In addition, the algorithm also computes the Jacobian of all the joints (\sa pinocchio::computeJointJacobians).
+  ///
+  /// \tparam JointCollection Collection of Joint types.
+  /// \tparam ConfigVectorType Type of the joint configuration vector.
+  ///
+  /// \param[in] model The model structure of the rigid body system.
+  /// \param[in] data The data structure of the rigid body system.
+  /// \param[in] q The joint configuration vector (dim model.nq).
+  /// \param[in] parentJointIdIn Index of the parent joint supporting the subtree.
+  /// \param[out] J A reference on the Jacobian matrix where the results will be stored in (dim 3 x model.nv). You must fill J with zero elements, e.g. J.fill(0.).
+  ///
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl, typename ConfigVectorType>
+  inline void
+  jacobianSubtreeCenterOfMass(const ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+                              DataTpl<Scalar,Options,JointCollectionTpl> & data,
+                              const Eigen::MatrixBase<ConfigVectorType> & q,
+                              const JointIndex & parentJointIdIn,
+                              typename DataTpl<Scalar,Options,JointCollectionTpl>::Matrix3x & JacobianOut);
```

Comment:
It would be better to use here: `const Eigen::MatrixBase<Matrix3xLike> & JacobianOut` to allow any Eigen type

Reply:
What do you mean by Matrix3xLike ? Isn't the code here templated by `Scalar` already ?

Reply:
You need to have an input which can represent any Matrix3 (which can be a dense matrix, a simple block, etc.). This why in all our algorithms, we have always something like `Eigen::MatrixBase<MatrixType> & ` which says that we can take as input any Eigen Matrix like type.

I encourage you yo read this excellent post in the Eigen documentation: https://eigen.tuxfamily.org/dox/TopicFunctionTakingEigenTypes.html

Reply:
You need to have an input which can represent any Matrix3 (which can be a dense matrix, a simple block, etc.). This why in all our algorithms, we have always something like `Eigen::MatrixBase<MatrixType> & ` which says that we can take as input any Eigen Matrix like type.

I encourage you to read this excellent post in the Eigen documentation: https://eigen.tuxfamily.org/dox/TopicFunctionTakingEigenTypes.html